crates/router/src/core/payment_methods/cards.rs

```diff
@@ -3550,6 +3554,11 @@ pub async fn list_payment_methods(
             business_profile.collect_billing_details_from_wallet_connector
         }));
 
+    let is_tax_connector_enabled = business_profile
+        .as_ref()
+        .map(|business_profile| business_profile.is_tax_connector_enabled)
+        .unwrap_or(false);
```

Comment:
Can we have a method on the business profile domain model that considers both the `is_tax_connector_enabled` field and the `tax_connector_id` field in the business profile, that returns a boolean indicating whether tax connector is enabled or not?

Reply:
@SanchithHegde added validation for tax_connector_id